### PR TITLE
Update MultiIndex checks

### PR DESCRIPTION
**Pull request by jbrockmendel**
cc @WillAyd I'm holding off on most non-trivial groupby stuff, LMK if this is a problem.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
@jreback @WillAyd gentle ping for this and #29456.  Neither is important on its own, but both will have merge conflicts with more-important PRs on their way.

File: pandas/_libs/reduction.pyx [hunk 1]
```diff
@@ -644,7 +644,7 @@ def compute_reduction(arr, f, axis=0, dummy=None, labels=None):
 
     if labels is not None:
         # Caller is responsible for ensuring we don't have MultiIndex
-        assert not labels._has_complex_internals
+        assert labels.nlevels == 1
```
Comment (WillAyd) [inline pandas/_libs/reduction.pyx]: Any reason not to do an isinstance check here? Not a strong blocker usage is mixed; might be nice to settle on one way of doing this

**Review by WillAyd — COMMENTED** [review]
Minor comment but not a big deal. OK with merging as is

File: pandas/_libs/reduction.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/reduction.pyx]: because we're in cython and dont want to import non-cython

**Review by WillAyd — APPROVED** [review]

**Comment by jreback** [comment]
can we completely remove _has_complex_internals? also IIRC there is an open PR about removing it (should close).

**Comment by jbrockmendel** [comment]
> can we completely remove _has_complex_internals? also IIRC there is an open PR about removing it (should close).

I pushed a rebase on to that this morning.  After this I'll probably have to rebase it again and thatll remove _has_complex_internals entirely.

**Comment by jreback** [comment]
lgtm. merge master once again as I *think* we merged something related to this in last 4 hours.

**Comment by jbrockmendel** [comment]
rebased+green

**Comment by jreback** [comment]
thxs

**Pull request merged**